packages/core-db/src/app/ethereum/ethereum-block-processor.ts

```diff
@@ -92,6 +95,36 @@ export class EthereumBlockProcessor {
     const block: Block = await provider.getBlock(blockNumber, true)
     log.debug(`Received block: [${JSON.stringify(block)}].`)
 
+    if (
+      this.confirmsUntilFinal > 1 &&
+      !!block.transactions &&
+      !!block.transactions.length
+    ) {
```

Comment:
Yeah, which is convenient because the only way to wait for confirms in ethers that I'm aware of is on transactions 😄 . Also, regardless of wait times, the blocks will always be handled in order as guaranteed by `QueuedPersistedProcessor<T>`